src/ucp/core/ucp_request.inl

```diff
@@ -807,4 +804,51 @@ ucp_request_invoke_uct_completion(ucp_request_t *req, ucs_status_t status)
     ucp_invoke_uct_completion(&req->send.state.uct_comp, status);
 }
 
+static UCS_F_ALWAYS_INLINE void
+ucp_request_complete_and_dereg_send(ucp_request_t *sreq, ucs_status_t status)
+{
+    ucp_request_send_generic_dt_finish(sreq);
+    ucp_request_send_buffer_dereg(sreq);
+    ucp_request_complete_send(sreq, status);
+}
+
+static UCS_F_ALWAYS_INLINE void
+ucp_request_complete_recv_rndv(ucp_request_t *req, ucs_status_t status)
```

Comment:
this is relevant only for RNDV request, right?

Reply:
yes